### DISCUSSION TITLE
Feedback for NeoAlgo

### DISCUSSION THREAD

**Discussion by HarshCasper** [TesseractCoding/NeoAlgo#1948] [General]
## Feedback for NeoAlgo 

Hello, Everyone 👋 

I'm Harsh, one of the Co-Founders of Tesseract Coding and the Maintainer of NeoAlgo. A few months back, we decided to kick-start this Open-Source Project, to help people get started with Open-Source. We saw a lack of issues that plagued the Open-Source Community. Few of them included: 

- Large Codebases which make new contributors intimidated. 
- Lots and Lots of Checks and CI configs that demotivated new contributors.
- Specific higher-bar of entry that didn't help beginner programmers and developers. 
- Lack of an Open-Source Community that can help people get started up. 

We combined all these problems, into a single solution which we know today as "NeoAlgo". Looking from plain sight, it is "just another" Data Structures and Algorithms repository. But the specific reason why we have nearly 200+ Contributors and 15+ Maintainers is that we believe that Open-Source is for everyone. 

And we wish to bring a Project that can help everyone get acquainted with how Open-Source Communities work and function. And over these few months, we have got a staggering number of contributions. And now, is the time, when we actually ask our Contributors and Maintainers on how are we actually doing. 

I would like to request all the Contributors and Maintainers to put a short note in the Discussion below about NeoAlgo. You don't have to give an "Oscar Speech" but it should be something that can be reassuring to new contributors that we welcome them in the Community. 

If there is something that we have done exceptional, please mention it. If there are some shortcomings, mention that specifically, because we are looking for increasing feedbacks on how we can improve. 

Every contribution counts! And your feedback would help us get a note of motivation to work towards the vision. 

Thank You!

**Comment by Abhijit2505** [comment]
**NeoAlgo** is a great initiative and I still remember the pace with which we receive contributions from our contributors.
Let's make it bigger.

**Comment by Pranav016** [comment]
It was great contributing to this repo under WoC! Learnt alot about industry best practices, how to write clean code and code that is easily understandable for everyone. I am sure this experience will help everybody in the future. Cheers✌️

**Comment by gat786** [comment]
I still can't imagine how you guys run such a big repo such efficiently. Kudos to all the maintainers. Great work overall.

**Comment by fomalhauting** [comment]
It was a great experience, I am glad to be a contributor.

**Comment by satyampgt4** [comment]
It was a great experience to be part of NeoAlgo. I learned a lot about open source contribution. Thank you NeoAlgo

**Comment by kartikp36** [comment]
It was my first open-source contribution through the WOC competition! I'm glad that I got a chance to contribute, mentors were very attentive, helpful and quick for the changes needed in the PR. Also, I like the idea of NeoAlgo's diverse great content for learning various kinds of Data Structures and Algorithms made in open-source way! Thank you Mentors and NeoAlgo for the chance and the efficient work maintaining the repo. :smiley:

**Comment by Bharati2612** [comment]
It was a great experience although I am new to open-source. WoC provided me this opportunity to start this journey. I am glad to be added as a contributor to NeoAlgo. Kudos to all the mentors and reviewers for their constant support and guidance. :)

**Comment by gargVader** [comment]
Overcame my fear of git and submitted as many as 17 PRs. Very helpful maintainers who guided me when I made any mistakes. Thank you

**Reply by aniketsingh98571** [reply]
It's great that you made 17 PR's to this repo.

**Reply by gat786** [reply]
Thank you for giving so much time to improve this repository 👏👏

**Comment by aniketsingh98571** [comment]
I get to know about TesseractCoding while I was selected for HackinCodes, it was then I raised my PR to this repo, and surprisingly it was my start of the beginning of contribution to the open-source journey, and it's still going on.

**Comment by barnamensch056** [comment]
It was a great experience working in this repo apart from coding get to know about formatting code in pep8 and clang formats from the mentors, looking forward to work on this repo again and thanks to WOC, get the certificate  too😁.

**Comment by asimarise** [comment]
This Project is amazing ✨🔥🔥 (Dev + DSA) I want to work on this
but where does it host I want to view it

**Comment by weirdrag08** [comment]
This is an amazing project and the hard work from the Project Admin's side are also appreciable. Thanks a lot for everything and looking forward to contribute to this repo more. ;)

**Comment by RaiyanMahin** [comment]
this project is a great resource for Data Structure and Algorithm learners.  I'm glad to contribute to this project.
Project Admins and Mentors are very helpful and hard working. 💯 
Looking forward to contribute more :)

**Comment by rutujaingole** [comment]
This is an awesome project for having all the Data Structures and Algorithms in various languages at one place!
Glad to be able to contribute given the responsive and active Mentors and Project Admins who are always there to help.
Looking forward to contribute more to this project.✨

**Comment by XZANATOL** [comment]
I like the fact that this project takes care of not only algorithms in general, but also many aspects in multiple fields. There is Competitive Programming, Backtracking, ML, and even Mathematics (my favourite :smile:), Moreover, Admins & Mentors here are actively available to help, make code readable for newcomers, and doing their best to achieve both user experience through automation and GitHub-Actions, and security through reviewing PRs, which makes the waiting period much less comparing with the number of contributions that come per day. :smile: 

Even if I finished GSSOC, I think I will be active here for a while, there are much to contribute to and learn from than it seemed from the first glance. :smile:

**Comment by batul02** [comment]
This is a great place for beginners and I am a GSSoC'21 participant and admire the Project Admin for this project @HarshCasper . He has be a great help to every participant.✨

**Comment by sumitgsh** [comment]
NeoAlgo project is a great place for contributors to land in. Documentation is well maintained and very easy to follow to get started which I mostly liked.😊😊
Contributing to this repo will help u learn how to write clean code and will also motivate you to learn DSA from the variety of resources available.
The maintainer are very engaging and will help u in every bit just have the patience after asking.

**Comment by Nikitha2309** [comment]
This is really a good project for those who want to focus on both open source and DSA😀.
I learned a lot of DSA, contributing to this project. 
I would like to thank @HarshCasper and all the mentors, for encouragement and for keeping the repo active.
Looking forward to **contribute more and more** and **be a member of this project**, Thank you.

**Comment by nagasrihith608** [comment]
Just wanted to take some time to appreciate and thank @HarshCasper for helping me understand and encourage me to work in open source! He has just planned everything right for us. Coming up with such incentives to work really motivates me. Thank you!!
